bigquery/google/cloud/bigquery/table.py

```diff
@@ -1081,6 +1072,46 @@ def _build_schema_resource(fields):
     return infos
 
 
+def _item_to_row(iterator, resource):
+    """Convert a JSON row to the native object.
+
+    .. note::
+
+        This assumes that the ``schema`` attribute has been
+        added to the iterator after being created, which
+        should be done by the caller.
+
+    :type iterator: :class:`~google.cloud.iterator.Iterator`
+    :param iterator: The iterator that is currently in use.
+
+    :type resource: dict
+    :param resource: An item to be converted to a row.
+
+    :rtype: tuple
+    :returns: The next row in the page.
+    """
+    return _row_from_json(resource, iterator.schema)
+
+
+# pylint: disable=unused-argument
```

Comment:
Yeah I understand, maybe it's not appropriate to discuss in this PR but I'm questioning if we should change the signature that the caller is...[calling](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/392f0f8b2ffaee98766f2851df72221051124100/core/google/cloud/iterator.py#L325) to make the iterator optional.  Just an idea to clean these up in the future.